modules/gdscript/gdscript.cpp

```diff
@@ -179,13 +180,14 @@ GDScriptInstance *GDScript::_create_instance(const Variant **p_args, int p_argco
 	if (initializer != nullptr) {
 		initializer->call(instance, p_args, p_argcount, r_error);
 		if (r_error.error != Callable::CallError::CALL_OK) {
+			String error_text = Variant::get_call_error_text(instance->owner, "_init", p_args, p_argcount, r_error);
 			instance->script = Ref<GDScript>();
 			instance->owner->set_script_instance(nullptr);
 			{
 				MutexLock lock(GDScriptLanguage::singleton->mutex);
 				instances.erase(p_owner);
 			}
-			ERR_FAIL_V_MSG(nullptr, "Error constructing a GDScriptInstance.");
+			ERR_FAIL_V_MSG(nullptr, "Error constructing a GDScriptInstance: " + error_text);
```

Comment:
Could do something like:
```suggestion
			if (r_error.error == Callable::CallError::CALL_ERROR_TOO_FEW_ARGUMENTS) {
				ERR_FAIL_V_MSG(nullptr, "Error constructing a GDScriptInstance: _init(...) requires parameters. Consider adding default values.");
			} else {
			ERR_FAIL_V_MSG(nullptr, "Error constructing a GDScriptInstance: " + error_text);
```
Or similar, unsure if it should be added to the above case as well, or how to word it

Reply:
I'm not sure if this is the best place to do it, as there are many ways a constructor can be called. This message would also trigger if the constructor accepts 2 arguments and was called with 1, which could be an actual user mistake when calling it.

Reply:
Agreed, keeping it simple then :)